extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -370,7 +370,7 @@ export class PublishDatabaseDialog {
 			}
 
 			// change the database inputbox value to the connection's database if there is one
-			if (this.connection.options.database) {
+			if (this.connection.options.database && this.connection.options.database !== constants.master) {
```

Comment:
Is this a concern for other system databases (msdb and tempdb)?

I don't think dacpacs can be deployed to either of those, but they should still probably be excluded from the UI

Reply:
Yeah it makes sense to not show system dbs in the publish dropdowns. Created #11226.

DacFx doesn't block publishing to them and SSDT shows them in the database dropdown for publishing, but SSMS doesn't allow publishing to system dbs.